### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by Subomi**
Did method checking for http methods and rewrote the callback pulling code. All tests pass!

### CODE REVIEW

**Comment by Subomi** [comment]
Sorry for pushing wrongly ... i only intended to push changes from quick fix - method checking & callback pulling

File: index.js [hunk 1]
```diff
@@ -30,35 +30,36 @@ function Paystack(key) {
...
-
-      // Get arguments in endpoint => {id} in customer/{id}
+	  
+      // Get arguments in endpoint e.g {id} in customer/{id} and pull
+      // out from array
       var argsInEndpoint = endpoint.match(/{[^}]+}/g);
+     
       if (argsInEndpoint) {
         l = argsInEndpoint.length;
+        l
```
Comment (kehers) [inline index.js]: @Subomi Can you run tests to be sure everything works? This line is wrong.

Comment (Subomi) [inline index.js]: Yes ... I ran test before pushing. I edit not edit that part of d code ... but  I mistakenly pushed d whole branch ... the 2 previous commits were not meant to be part of d push

Comment (Subomi) [inline index.js]: I'll check it and run tests again.

**Pull request closed without merging**